providers/mimetypes.rb

```diff
@@ -0,0 +1,68 @@
+#
+## Cookbook Name:: gecos-ws-mgmt
+## Provider:: pdf_viewer
+##
+## Copyright 2013, Junta de Andalucia
+## http://www.juntadeandalucia.es/
+##
+## All rights reserved - EUPL License V 1.1
+## http://www.osor.eu/eupl
+##
+
+require 'etc'
+require 'chef/mixin/shell_out'
+include Chef::Mixin::ShellOut
+
+action :setup do
+	begin
+	  if new_resource.support_os.include?($gecos_os)
+    	package "xdg-utils" do
+       action :nothing
+      end.run_action(:install)
+
+      users = new_resource.users
+
+      users.each_key do |user_key|
+        nameuser = user_key
+        username = nameuser.gsub('###','.')
+        user = users[user_key]
+				gid = Etc.getpwnam(username).gid
+				xdg_code = ''
+
+				directory "/home/#{username}/.local/share/applications" do
+					owner username
+					group gid
+					recursive true
+				end
+
+				Chef::Log.debug("mimetypes.rb - Users: #{user}")
+
+				user.mimetyperelationship.each do |assoc|
```

Comment:
Buenas, Alfonso:

Con respecto a tus comentarios:

1.- He introducido en el código un warning para cuando no existe el
.desktop.
2.- He sustituido el recurso "bash" por "execute".

Me asalta una duda con tu último comentario:

"Missed check before acting. An xdg-mime query default mimetype will give
you the current association"

¿ Por qué querría saberlo ? Esa consulta te dice la aplicación asociada en
ese momento con el tipo mime. Pero la vas a sustituir por la nueva
configuración en la receta o se quiere preservar ? No entiendo.

Saludos.

2016-07-26 13:40 GMT+02:00 Alfonso E.M. <notifications@github.com>:

> In providers/mimetypes.rb
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145#discussion_r72232617>
> :
>
> > +      users.each_key do |user_key|
> > +        nameuser = user_key
> > +        username = nameuser.gsub('###','.')
> > +        user = users[user_key]
> > +				gid = Etc.getpwnam(username).gid
> > +				xdg_code = ''
> > +
> > +				directory "/home/#{username}/.local/share/applications" do
> > +					owner username
> > +					group gid
> > +					recursive true
> > +				end
> > +
> > +				Chef::Log.debug("mimetypes.rb - Users: #{user}")
> > +
> > +				user.mimetyperelationship.each do |assoc|
>
> Missed check before acting. An xdg-mime query default mimetype will give
> you the current association.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gecos-team/gecos-workstation-management-cookbook/pull/145/files/c25d460433d1f0d07780a75a6172e8a068ae51dd#r72232617>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATHDTMgWE8Kw_9M5NbBZkugjYfCFUmOaks5qZfI2gaJpZM4JRrKY>
> .
>